pints/_nested/__init__.py

```diff
@@ -780,13 +795,191 @@ def _update_logger(self):
             self._i_message += 1
             if self._i_message >= self._next_message:
                 # Log state
-                self._logger.log(self._i_message, self._sampler._n_evals,
-                                 self._timer.time(), self._diff,
-                                 float(self._sampler._accept_count /
-                                       (self._sampler._n_evals -
-                                        self._sampler._n_active_points)))
+                if not self._sampler._multiple_ellipsoids:
+                    self._logger.log(self._i_message, self._sampler._n_evals,
+                                     self._timer.time(), self._diff,
+                                     float(self._sampler._accept_count /
+                                           (self._sampler._n_evals -
+                                            self._sampler._n_active_points)))
+                else:
+                    self._logger.log(self._i_message, self._sampler._n_evals,
+                                     self._timer.time(), self._diff,
+                                     float(self._sampler._accept_count /
+                                           (self._sampler._n_evals -
+                                            self._sampler._n_active_points)),
+                                     self._sampler._ellipsoid_count)
 
                 # Choose next logging point
                 if self._i_message > self._message_warm_up:
                     self._next_message = self._message_interval * (
                         1 + self._i_message // self._message_interval)
+
+
+class Ellipsoid():
+    """
+    A class to represent N dimensional ellipsoids, which are needed by both
+    ellipsoidal nested sampling and MultiNest.
+
+    In "center form" the equation of an ellipsoid is given by:
+
+    ``(x-c).T * A * (x-c) = 1``
+
+    where ``A`` is a NxN dimensional positive definite symmetric matrix and
+    ``c`` is a N dimensional vector indicating the center of the ellipsoid.
+
+    Parameters
+    ----------
+    A : NxN dimensional positive definite symmetric matrix
+        represents the orientation and size of ellipsoid
+    c : N dimensional vector
+        center of ellipsoid
+    """
+    def __init__(self, A, c):
+
+        self._c = c
+        self._n_parameters = len(c)
+
+        if A.shape != (self._n_parameters, self._n_parameters):
+            raise ValueError(
+                'Sigma must have same dimension as mean, or be a square ' +
+                'matrix with the same dimension as the center.')
+        self._A = np.copy(A)
+
+        # calculate useful quantities
+        self._A_inv = np.linalg.inv(A)
+        # don't calculate volume unless needed
+        self._volume = None
+
+        # don't cache points unless constructed using minimum_volume_ellipsoid
+        self._points = None
+        self._n_points = 0
+
+    def __add__(self, other):
```

Comment:
You're defining a very unusual algebra here, where x + y is an object outside of the space x and y live in...

Reply:
Also... is `[a, b]` too much effort to write versus `a + b` ?

Reply:
Yep, but I wasn't sure how to handle combining lots of ellipsoids. Specifically, I wasn't sure how to make `leaf_ellipsoids` work using lists. I also thought it has a nice elegance which has a nice analogy to `n_leaf_ellipsoids`?

I take the point about the addition resulting in a different object though I'm not sure people would necessarily confuse it with a well-defined algebra...

Reply:
What about a + b + c though, that'd become either [[a, b], c] or [a, [b, c]] deciding on which order python makes the calls in?

Reply:
So, hilariously, despite much energy and thought on my side, I wasn't actually using this add function. I thought I was but actually I was just adding lists! (So, I've deleted it.)